src/pipelinedefinition.cpp

```diff
@@ -411,13 +423,29 @@ class NodeValidator {
                 shape.size());
             return StatusCode::PIPELINE_NOT_ENOUGH_SHAPE_DIMENSIONS_TO_DEMULTIPLY;
         }
-        if (shape[1] != demultiplicatorNodeInfo.demultiplyCount.value()) {
-            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline: {} definition failed. Demultiply count: {} of node: {} does not match tensor second dimenson value: {}",
+        if (demultiplicatorNodeInfo.demultiplyCount.value() != 0) {
+            if (shape[1] != 0) {
+                // 0 means that node accepts dynamic shape
+                if (shape[1] != demultiplicatorNodeInfo.demultiplyCount.value()) {
+                    SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline: {} definition failed. Demultiply count: {} of node: {} does not match tensor second dimenson value: {}",
+                        this->pipelineName,
+                        demultiplicatorNodeInfo.demultiplyCount.value(),
+                        demultiplicatorNodeInfo.nodeName,
+                        shape[1]);
+                    return StatusCode::PIPELINE_DEMULTIPLY_COUNT_DOES_NOT_MATCH_BLOB_SHARD_COUNT;
+                }
+            } else {
+                SPDLOG_LOGGER_WARN(modelmanager_logger, "Demultiply count: {} of node: {} is fixed while second dimenson value of node library is not: {}. This pipeline may fail at execution stage.",
+                    demultiplicatorNodeInfo.demultiplyCount.value(),
+                    demultiplicatorNodeInfo.nodeName,
+                    shape[1]);
+            }
+        } else if (shape[1] != 0) {
+            SPDLOG_LOGGER_WARN(modelmanager_logger, "Demultiply count: {} of node: {} is dynamic while second dimenson value of gather node is not: {}. This pipeline may fail at execution stage.",
```

Comment:
"of gather node"? Shouldn't it be "demultiplexer node"?

Reply:
Should it really be a warning? Administrator asks pipeline to be dynamic, but node always produce static batch, I think this meets criteria of an administrator.

Reply:
This warning is for case when you have dynamic demultiplexer but static gatherer. It may happen that this will work due to other circumstances but OVMS does not have knowledge that the dynamic demultiplexer will produce exactly the batch expected by the gatherer.

There are two tests for those two WARN logs. They do not test against warn log, but if you run those tests you will see the warnings.